ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/services/admin/ApiPartyManagementService.scala

```diff
@@ -47,14 +48,51 @@ class ApiPartyManagementService private (
       .listParties()
       .map(ps => ListKnownPartiesResponse(ps.map(mapPartyDetails)))(DE)
 
+  /**
+    * Continuously polls tha party management service to check if a party has been persisted.
+    *
+    * Blocking.
+    *
+    * @param party The party whose persistence we're waiting for
+    * @return The number of attempts before the party was found
+    */
+  private def pollUntilPersisted(party: String): Int =
+    Iterator
+      .from(1)
+      .find { attempt =>
+        logger.debug(s"Waiting for party '$party' to be persisted (attempt #$attempt)...")
+        partyManagementService
+          .listParties()
+          .map {
+            case persisted if persisted.exists(_.party == party) => true
+            case _ => false
+          }(DE)
+        Await.result(
+          partyManagementService
+            .listParties()
+            .map {
+              case persisted if persisted.exists(_.party == party) => true
+              case _ => false
+            }(DE),
+          Duration.Inf)
+      }
+      .get
+
+  private def pollUntilPersisted(result: AllocatePartyResponse): Future[AllocatePartyResponse] = {
+    require(result.partyDetails.isDefined, "Party allocation response must have the party details")
+    Future(pollUntilPersisted(result.partyDetails.get.party))(DE).map { numberOfAttempts =>
```

Comment:
`pollUntilPersisted` shoud probably just return the `Future` (and not use `Await.result`) if we anyway are going to put it into a `Future` again.

Reply:
Addressed from 621652d5e